plugins/inputs/x509_cert/x509_cert.go

```diff
@@ -446,6 +452,24 @@ func (c *X509Cert) getCert(u *url.URL, timeout time.Duration) ([]*x509.Certifica
 		ocspresp := tlsConn.ConnectionState().OCSPResponse
 
 		return certs, &ocspresp, nil
+	case "jks":
+		fileType, err := detectKeystoreFormat(u.Path)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		switch fileType {
+		case "jks":
+			c.Log.Tracef("Entering JKS for: %v", u.Path)
+			certs, err := c.processJKS(u.Path)
+			return certs, nil, err
+		case "pkcs12":
+			c.Log.Tracef("Entering PKCS#12 for: %v", u.Path)
+			certs, err := c.processPKCS12(u.Path)
+			return certs, nil, err
+		default:
+			return nil, nil, fmt.Errorf("unknown keystore format for %v", u.Path)
+		}
```

Comment:
To be honest, too much magic is going on here! Let the user specify what the URI is about and don't try to be clever... It has been a major fault in the past to _not_ enforce the `file://` prefix causing trouble and complicates code...
```suggestion
	case "jks":
		certs, err := c.processJKS(u.Path)
		return certs, nil, err
	case "pkcs12":
		certs, err := c.processPKCS12(u.Path)
		return certs, nil, err
```

Reply:
Make sense to me! I prefer this approach.

My initial approach was to reduce the user definition, however, that can create assumptions and lead to issues, so I prefer your approach. Fixed